From: Lorenzo Stoakes <lorenzo.stoakes@oracle.com>
Subject: fixup io_remap_pfn_range_[prepare, complete]
Date: Thu, 18 Sep 2025 07:43:21 +0100

propagate the fact that we don't need io_remap_pfn_range_prot()

Link: https://lkml.kernel.org/r/2cf129c4-627b-4a78-9ec3-cf43c95cf17d@lucifer.local
Signed-off-by: Lorenzo Stoakes <lorenzo.stoakes@oracle.com>
Signed-off-by: Andrew Morton <akpm@linux-foundation.org>
---

 mm/internal.h |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/mm/internal.h~mm-introduce-io_remap_pfn_range_-fix
+++ a/mm/internal.h
@@ -1674,7 +1674,7 @@ static inline int io_remap_pfn_range_com
 		pgprot_t orig_prot)
 {
 	const unsigned long pfn = io_remap_pfn_range_pfn(orig_pfn, size);
-	const pgprot_t prot = io_remap_pfn_range_prot(orig_prot);
+	const pgprot_t prot = pgprot_decrypted(orig_prot);
 
 	return remap_pfn_range_complete(vma, addr, pfn, size, prot);
 }
_
